paasta_tools/firewall_logging.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import logging
+
+import syslogmp
+from six.moves import socketserver
+
+from paasta_tools.firewall import services_running_here
+from paasta_tools.utils import _log
+from paasta_tools.utils import configure_log
+from paasta_tools.utils import load_system_paasta_config
+
+
+log = logging.getLogger(__name__)
+
+
+class SyslogUDPHandler(socketserver.BaseRequestHandler):
+    def setup(self):
+        configure_log()
+        self.cluster = load_system_paasta_config().get_cluster()
+
+    def handle(self):
+        data, socket = self.request
+        syslog_to_paasta_log(data, self.cluster)
+
+
+def syslog_to_paasta_log(data, cluster):
+    iptables_log = parse_syslog(data)
+    if iptables_log is None:
+        return
+
+    service, instance = lookup_service_instance_by_ip(iptables_log['SRC'])
+    if service is None or instance is None:
+        return
+
+    # prepend hostname
+    log_line = iptables_log['hostname'] + ': ' + iptables_log['message']
```

Comment:
agreed that it would be cool to eventually transform this into something more readable (maybe looking up smartstack ports), but this is a great start